testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java

```diff
@@ -421,15 +477,80 @@ public void testReferencedTablesWithMaterializedViews()
+    }
+
+    // Currently, the storage table for a materialized view is not included anywhere in the set of `tables` in the query event.
+    // See for more details: https://github.com/trinodb/trino/pull/18871#discussion_r1412247513
+    @Test
+    public void testReferencedTablesWithMaterializedViewsFresh()
+            throws Exception
+    {
+        QueryEvents queryEvents = runQueryAndWaitForEvents("SELECT nationkey FROM mock.default.test_materialized_view_fresh").getQueryEvents();
+
+        QueryCompletedEvent event = queryEvents.getQueryCompletedEvent();
+
+        List<TableInfo> tables = event.getMetadata().getTables();
+        assertThat(tables).hasSize(2);
+        TableInfo table = tables.get(0);
+        assertThat(table)
+                .hasCatalogSchemaTable("tpch", "tiny", "nation")
+                .hasAuthorization("alice")
+                .isDirectlyReferenced(false)
+                .hasColumnsWithoutMasking("nationkey", "regionkey", "name", "comment")
+                .hasFilters()
+                .hasTableReferencesSatisfying(tblRef -> assertThat(tblRef).asViewInfo(true).hasCatalogSchemaView("mock", "default", "test_materialized_view_fresh"));
+
+        table = tables.get(1);
+        assertThat(table)
+                .hasCatalogSchemaTable("mock", "default", "test_materialized_view_fresh")
+                .hasAuthorization("user")
+                .isDirectlyReferenced(true)
+                .hasColumnsWithoutMasking("nationkey")
+                .hasFilters()
+                .hasTableReferencesSatisfying()
+                .hasViewText("SELECT * FROM tpch.tiny.nation");
+    }
+
+    @Test
+    public void testReferencedTablesWithViewsAndRedirection()
+            throws Exception
+    {
+        QueryEvents queryEvents = runQueryAndWaitForEvents("SELECT test_column FROM mock.default.test_view_with_redirect").getQueryEvents();
+
+        QueryCompletedEvent event = queryEvents.getQueryCompletedEvent();
+
+        List<TableInfo> tables = event.getMetadata().getTables();
+        assertThat(tables).hasSize(2);
+
+        TableInfo table = tables.get(0);
+        assertThat(table)
+                .hasCatalogSchemaTable("tpch", "tiny", "nation")
+                .hasAuthorization("user")
+                .isDirectlyReferenced(false)
+                .hasColumnsWithoutMasking("nationkey")
+                .hasFilters()
+                .hasTableReferencesSatisfying(tblRef -> assertThat(tblRef).asViewInfo(false).hasCatalogSchemaView("mock", "default", "test_view_with_redirect"));
+
+        table = tables.get(1);
+        assertThat(table)
+                .hasCatalogSchemaTable("mock", "default", "test_view_with_redirect")
+                .hasAuthorization("user")
+                .isDirectlyReferenced(true)
+                .hasColumnsWithoutMasking("test_column")
+                .hasFilters()
+                .hasTableReferencesSatisfying();
```

Comment:
We are not passing any arguments. Is it intentional ?

Reply:
Yes ... It is asserting that there are no filters, and no table references, respectively. I agree it's a little confusing, let me update.